packages/write-project-manifest/src/index.ts

```diff
@@ -1,25 +1,33 @@
 import { ProjectManifest } from '@pnpm/types'
-import writeJsonFile = require('write-json-file')
-import writeJson5File = require('write-json5-file')
+import JSON5 = require('json5')
+import fs = require('mz/fs')
+import path = require('path')
+import writeFileAtomic = require('write-file-atomic')
 import writeYamlFile = require('write-yaml-file')
 
 const YAML_FORMAT = {
   noCompatMode: true,
   noRefs: true,
 }
 
-export default function writeProjectManifest (
+export default async function writeProjectManifest (
   filePath: string,
   manifest: ProjectManifest,
-  opts?: { indent?: string | number | undefined }
+  opts?: {
+    indent?: string | number | undefined,
+    insertFinalNewline?: boolean,
+  }
 ): Promise<void> {
-  switch (filePath.substr(filePath.lastIndexOf('.') + 1).toLowerCase()) {
-    case 'json5':
-      return writeJson5File(filePath, manifest, opts)
-    case 'yaml':
-      return writeYamlFile(filePath, manifest, YAML_FORMAT)
-    case 'json':
-    default:
-      return writeJsonFile(filePath, manifest, opts)
+  const fileType = filePath.substr(filePath.lastIndexOf('.') + 1).toLowerCase()
+  if (fileType === 'yaml') {
+    return writeYamlFile(filePath, manifest, YAML_FORMAT)
   }
+
+  await fs.mkdir(path.dirname(filePath), { recursive: true })
+  const trailingNewline = opts?.insertFinalNewline === false ? '' : '\n'
+
+  const json = (fileType === 'json5' ? JSON5 : JSON)
+    .stringify(manifest, null, opts?.indent ?? '\t')
```

Comment:
👍🏻 on using [the more accessible tabs](https://www.reddit.com/comments/c8drjo) by default.